### PR TITLE
CHERI-256 tidying

### PR DESCRIPTION
**Pull request by brooksdavis**
A couple more 256-bit removals.

### CODE REVIEW

File: share/mk/bsd.cpu.mk [hunk 1]
```diff
@@ -348,13 +348,8 @@ CFLAGS+=	-Werror=implicit-function-declaration
 # XXXBD: is this still needed?
 CFLAGS+=	-Wno-deprecated-declarations
 # XXXBD: is -mstack-alignment needed here?
-. if ${MACHINE_ARCH:Mmips*c128}
 CFLAGS+=	-cheri=128
 CFLAGS+=	-mstack-alignment=16
```
Comment (jrtc27) [inline share/mk/bsd.cpu.mk:352]: We can drop this too (and the comment just above); n64 requires 16-byte quadword alignment, only CHERI-256 ever needed to change the ABI there.

**Comment by jrtc27** [comment]
You've got a stray duplicate NFSv4 commit; rebase on dev?

File: share/mk/bsd.cpu.mk [hunk 2]
```diff
@@ -347,14 +347,7 @@ CFLAGS+=	-Werror=implicit-function-declaration
 # Turn off deprecated warnings
 # XXXBD: is this still needed?
 CFLAGS+=	-Wno-deprecated-declarations
-# XXXBD: is -mstack-alignment needed here?
-. if ${MACHINE_ARCH:Mmips*c128}
 CFLAGS+=	-cheri=128
```
Comment (arichardson) [inline share/mk/bsd.cpu.mk:350]: We could also change this to `-cheri` since 128 is the default.

File: lib/libprocstat/core.c [hunk 3]
```diff
@@ -70,9 +70,6 @@ __FBSDID("$FreeBSD$");
 #ifndef EF_MIPS_MACH_CHERI128
 #define	EF_MIPS_MACH_CHERI128	0x00C10000
 #endif
-#ifndef	EF_MIPS_MACH_CHERI256
-#define EF_MIPS_MACH_CHERI256	0x00C20000
-#endif
 #ifndef EF_MIPS_MACH
 #define	EF_MIPS_MACH		0x00FF0000
 #endif
```
Comment (arichardson) [inline lib/libprocstat/core.c:75]: Shouldn't all of these already be defined by sys/elf.h?

Comment (jrtc27) [reply lib/libprocstat/core.c:75]: Do gelf.h/libelf.h end up pulling that in or just define their own set of types and macros?

Comment (bsdjhb) [reply lib/libprocstat/core.c:75]: gelf.h and libelf.h have their own macros, but I'd rather us patch those over in contrib and not need the #ifndef here, but that probably warrants a separate cleanup.

**Pull request merged**